art/art_logger.py

```diff
@@ -0,0 +1,73 @@
+from contextlib import contextmanager
+import sys
+from loguru import logger
+from pathlib import Path
+from art.paths import LOG_PATH
+logger.remove()
+logger.add(sys.stdout, format="{message}", level="DEBUG", filter= lambda record: ('halt_exception' not in record['extra']) or (not record['extra']['halt_exception']))
+
+
+class ArtLogInterceptor:
+    """
+    A class that intercepts writes to stdout and logs them using a logger.
+    """
+
+    def __init__(self, log_file, to_intercept, supress=False):
+        self.terminal = to_intercept
+        self.supress = supress
+        self.log = open(log_file, "a")
+
+    def write(self, message):
+        if not self.supress:
+            self.terminal.write(message)
+        self.log.write(message)
+
+    def flush(self):
+        if not self.supress:
+            self.terminal.flush()
+        self.log.flush()
+
+
+logger_level = 0
+already_contextualized_logers = set()
+
+
+@contextmanager
+def logger_contextualize(folder_path: Path, supress_stdout=False):
+    if str(folder_path) in already_contextualized_logers:
+        yield
+        return
+    already_contextualized_logers.add(str(folder_path))
+    global logger_level
```

Comment:
shouldn't this be more `number_of_loggers`?

Reply:
removed